crates/opencascade/src/mesh.rs

```diff
@@ -12,22 +15,20 @@ pub struct Mesh {
 }
 
 pub struct Mesher {
-    inner: UniquePtr<ffi::BRepMesh_IncrementalMesh>,
+    pub(crate) inner: UniquePtr<ffi::BRepMesh_IncrementalMesh>,
 }
 
 impl Mesher {
-    pub fn new(shape: &Shape) -> Self {
-        let inner = ffi::BRepMesh_IncrementalMesh_ctor(&shape.inner, 0.01);
+    pub fn try_new(shape: &Shape, triangulation_tolerance: f64) -> Result<Self, Error> {
+        let inner = ffi::BRepMesh_IncrementalMesh_ctor(&shape.inner, triangulation_tolerance);
 
-        if !inner.IsDone() {
-            // TODO(bschwind) - Add proper Error type and return Result.
-            panic!("Call to ffi::BRepMesh_IncrementalMesh_ctor failed");
+        match inner.IsDone() {
+            true => Ok(Self { inner }),
+            false => Err(Error::TriangulationFailed),
```

Comment:
I have a slight preference for `if` statements when it's a simple true/false match but this is a minor detail.

Reply:
Done.